### PR TITLE
feat(report): add support of GH Actions as VCS/report type

### PR DESCRIPTION
**Pull request by k-dovgan**

### CODE REVIEW

File: universum/modules/vcs/github_app_vcs.py [hunk 1]
```diff
@@ -128,21 +128,21 @@ def get_token(self, installation_id=None):
         return super().get_token(installation_id=self.settings.installation_id)
 
 
-class GithubMainVcs(ReportObserver, git_vcs.GitMainVcs, GithubTokenWithInstallation):
+class GithubAppMainVcs(ReportObserver, git_vcs.GitMainVcs, GithubTokenWithInstallation):
     """
     This class mostly contains functions for Gihub report observer
```
Comment (miltolstoy) [inline universum/modules/vcs/github_app_vcs.py:133]: "Gihub" -> "Github"

File: universum/modules/vcs/github_actions_vcs.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+from ...lib.gravity import Dependency
+
+__all__ = [
+    "GithubActionsMainVcs"
+]
+
+
+class GithubActionsMainVcs(ReportObserver, git_vcs.GitMainVcs):
+    """
+    This class mostly contains functions for Gihub report observer
```
Comment (miltolstoy) [inline universum/modules/vcs/github_actions_vcs.py:16]: "Gihub" -> "Github"

File: universum/modules/vcs/github_actions_vcs.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+        self.payload = self.read_and_check_multiline_option("payload", """
+            GitHub web-hook payload JSON is not specified.
+
+            Please pass incoming web-hook request payload to this parameter directly via
+            '--github-payload' ('-ghp') command line parameter or by setting GITHUB_PAYLOAD
+            environment variable, or by passing file path as the argument value (start
+            filename with '@' character, e.g. '@/tmp/file.json' or '@payload.json' for
+            relative path starting at current directory). Please note, that when passing
+            a file, it's expected to be in UTF-8 encoding
```
Comment (miltolstoy) [inline universum/modules/vcs/github_actions_vcs.py:53]: The final dot is missing.

File: universum/modules/vcs/github_actions_vcs.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+            """)
+
+        try:
+            self.payload_json = json.loads(self.payload)
+            self.settings.repo = self.payload_json['repository']['html_url']
+            self.settings.refspec = self.payload_json['pull_request']['head']['ref']
+        except json.decoder.JSONDecodeError as error:
```
Comment (miltolstoy) [inline universum/modules/vcs/github_actions_vcs.py:60]: Hande the `KeyError` also, as `self.payload` contains user input.
There are same issues with user-provided JSON usage without any checks below. Use the `.get()` method or exception handling depending on the situation.

File: universum/modules/vcs/github_actions_vcs.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+    def code_report_to_review(self, report):
+        # git show returns string, each file separated by \n,
+        # first line consists of commit id and commit comment, so it's skipped
+        commit_files = self.repo.git.show("--name-only", "--oneline", self.settings.checkout_id).split('\n')[1:]
+        # NB! When using GITHUB_TOKEN, the rate limit is 1,000 requests per hour per repository.
+        # (https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28  ->
+        #                                              #rate-limits-for-requests-from-github-actions)
+        for path, issues in report.items():
+            if path in commit_files:
```
Comment (miltolstoy) [inline universum/modules/vcs/github_actions_vcs.py:102]: Revert condition to decrease nesting level.

**Review by miltolstoy — COMMENTED** [review]
LGTM

File: universum/modules/vcs/github_app_vcs.py [hunk 6]
```diff
@@ -220,12 +220,13 @@ def code_report_to_review(self, report):
         comments = []
         for path, issues in report.items():
             if path in commit_files:
-                for issue in issues:
-                    comments.append(dict(path=path,
-                                         message=issue['message'],
-                                         start_line=issue['line'],
-                                         end_line=issue['line'],
-                                         annotation_level="warning"))
+                break
```
Comment (i-keliukh) [inline universum/modules/vcs/github_app_vcs.py:223]: I think the break here is incorrect. It stops processing of report as soon as issue in file on review is reached.

File: universum/modules/vcs/github_actions_vcs.py [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+    def __init__(self, *args, **kwargs) -> None:
+        self.settings.repo = "Will be filled after payload parsing"
+        super().__init__(*args, **kwargs)
+        self.reporter: Reporter = None  # type: ignore
+
+        self.check_required_option("token", """
+            The GitHub workflow token is not specified.
+
+            For github the git checkout id defines the commit to be checked and reported.
```
Comment (i-keliukh) [inline universum/modules/vcs/github_actions_vcs.py:41]: The error message is incorrect

File: universum/modules/vcs/github_actions_vcs.py [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+        }
+
+        utils.make_request(url, request_method="POST", json=request, headers=headers, timeout=5*60)
+
+    def code_report_to_review(self, report: dict) -> None:
+        # git show returns string, each file separated by \n,
+        # first line consists of commit id and commit comment, so it's skipped
+        commit_files: List[str] = self.repo.git.show("--name-only", "--oneline",
+                                                     self.settings.checkout_id).split('\n')[1:]
+        # NB! When using GITHUB_TOKEN, the rate limit is 1,000 requests per hour per repository.
```
Comment (i-keliukh) [inline universum/modules/vcs/github_actions_vcs.py:104]: It is not clear why this comment is here. It doesn't seem to relate to any code around.

Comment (k-dovgan) [reply universum/modules/vcs/github_actions_vcs.py:103]: Added explanation.

**Review by i-keliukh — APPROVED** [review]

**Review by i-keliukh — DISMISSED** [review]

**Review by i-keliukh — APPROVED** [review]

**Pull request merged**